From 3eb6cae898bb6c50a738474e6f97756acb33d44e Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan@agner.ch>
Date: Tue, 29 Dec 2020 23:34:52 +0100
Subject: [PATCH 1/3] rpi: add NVMe to boot order

The Compute Module 4 I/O Board can support a NVMe. Add NVMe to the boot
order.

Patch source:
https://github.com/home-assistant/operating-system/tree/dev/buildroot-external/board/raspberrypi/patches/uboot
at commit SHA b0bbe70315ee6474637b6c4242c41909c9ec2db8.

Upstream-Status: Backport
Signed-off-by: Stefan Agner <stefan@agner.ch>
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/rpi.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index b158eee903..391b8aa2dd 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -149,6 +149,12 @@
 	#define BOOT_TARGET_MMC(func)
 #endif
 
+#if CONFIG_IS_ENABLED(CMD_NVME)
+	#define BOOT_TARGET_NVME(func) func(NVME, nvme, 0)
+#else
+	#define BOOT_TARGET_NVME(func)
+#endif
+
 #if CONFIG_IS_ENABLED(CMD_USB)
 	#define BOOT_TARGET_USB(func) func(USB, usb, 0)
 #else
@@ -170,6 +176,7 @@
 #define BOOT_TARGET_DEVICES(func) \
 	BOOT_TARGET_MMC(func) \
 	BOOT_TARGET_USB(func) \
+	BOOT_TARGET_NVME(func) \
 	BOOT_TARGET_PXE(func) \
 	BOOT_TARGET_DHCP(func)
 
-- 
2.17.1

